### ISSUE TITLE
Point Cloud Texture changes (bugs out) through Activity lifecycles

### ISSUE THREAD

**Issue by 544531645** [viromedia/virocore#85]
When I loaded the plane and move phone ,the track 3d model  can't quick follow ,and it seems the phone very slow .

**Comment by 544531645** [comment]
public void  init(){

        if(mScene !=null) {
            mScene.setListener(null);
            mScene.getRootNode().removeAllChildNodes();
            mScene.getRootNode().removeAllLights();
            mScene.getRootNode().removeAllSounds();
            ((ViroViewARCore)mViroView).setCameraARHitTestListener(null);
            mScene.resetPointCloudSurface();
            controller.surfaces.clear();
            mCrosshairModel= null;
            mScene.dispose();
            mScene = null;
        }
        mScene = new ARScene();
        mScene.displayPointCloud(true);


        // Create an TrackedPlanesController to visually display tracked planes
        controller = new TrackedPlanesController(getActivity() ,mViroView);

        // Spawn a 3D Droid on the position where the user has clicked on a tracked plane.
        controller.addOnPlaneClickListener(new ClickListener() {
            @Override
            public void onClick(int i, Node node, Vector clickPosition) {
                //createDroidAtPosition(mCrosshairModel.getPositionRealtime());
            }

            @Override
            public void onClickState(int i, Node node, ClickState clickState, Vector vector) {
                //No-op
            }
        });

        mScene.setListener(controller);

        mViroView.setScene(mScene);
        ARHitTestListenerCrossHair mCrossHairHitTest = new ARHitTestListenerCrossHair();
        ((ViroViewARCore)mViroView).setCameraARHitTestListener(mCrossHairHitTest);
        AmbientLight  mMainLight = new AmbientLight(Color.parseColor("#606060"), 2500);
        mMainLight.setInfluenceBitMask(3);

        mScene.getRootNode().addLight(mMainLight);
        initARCrossHair(mScene);
        init3DModelProduct(mScene);
        Drawable drawable = getActivity().getResources().getDrawable(R.drawable.point_colud);
        Bitmap bitmap = drawableToBitmap(drawable);
        Texture chineseCharTexture  = new Texture(bitmap, Texture.Format.RGBA8, false, true);
        Material material1 = new Material();
        material1.setDiffuseTexture(chineseCharTexture);
        Surface surface = new Surface(0.25f, 0.25f);
        surface.setMaterials(Arrays.asList(material1));

        mScene.setPointCloudSurface(surface);
    }

**Comment by 544531645** [comment]
I changed the color of point but occasionally it was white. Drawable drawable = getActivity().getResources().getDrawable(R.drawable.point_colud);
    Bitmap bitmap = drawableToBitmap(drawable);
    Texture chineseCharTexture  = new Texture(bitmap, Texture.Format.RGBA8, false, true);
    Material material1 = new Material();
    material1.setDiffuseTexture(chineseCharTexture);
    Surface surface = new Surface(0.25f, 0.25f);
    surface.setMaterials(Arrays.asList(material1));

    mScene.setPointCloudSurface(surface);

**Comment by 544531645** [comment]
I use this do full scrren .. but in viroview I don't know why somtime it not work.                                                        
   @Override
    public void onWindowFocusChanged(boolean hasFocus) {
        super.onWindowFocusChanged(hasFocus);
        if (hasFocus && Build.VERSION.SDK_INT >= 21) {
            View decorView = getWindow().getDecorView();
            decorView.setSystemUiVisibility(
                    View.SYSTEM_UI_FLAG_LAYOUT_STABLE
                            | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
                            | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
                            | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
                            | View.SYSTEM_UI_FLAG_FULLSCREEN
                            | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY);
        }
    }

**Comment by VikAdvani** [comment]
Hi @544531645,
1)For the 1st issue, the slow tracking, what device are you using? What Android OS is it?
2)For the 2nd issue, how often do you see the color of the point cloud turn white? Do all the dots turn white or just some of them? Can you send a screenshot if possible?
3)For the 3rd issue, the correct way to override the system UI Visibility flags is to do the following:
Create a class that inherits from ViroViewARCore and override the getSystemUIVisibilityFlags method:


```
 @Override
 protected int getSystemUiVisibilityFlags() {
 return (View.SYSTEM_UI_FLAG_LAYOUT_STABLE
                | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN);
}

**Comment by 544531645** [comment]
1 . android pixel 
2.
![32294245381923843](https://user-images.githubusercontent.com/15622364/38969050-e1ebb688-43c0-11e8-8451-ab2094b63b2c.jpg)
![122275133336257028](https://user-images.githubusercontent.com/15622364/38969052-e3ae17ae-43c0-11e8-8955-6741a4a67896.jpg)

**Comment by 544531645** [comment]
open twice  it is different

**Comment by 544531645** [comment]
3.getSystemUiVisibilityFlags this method at where setSystemUiVisibilityFlags

**Comment by 544531645** [comment]
![853707981770492252](https://user-images.githubusercontent.com/15622364/38969314-34b89c90-43c2-11e8-9d26-90ef0d0c8b49.jpg)
![412299719645957852](https://user-images.githubusercontent.com/15622364/38969316-3622ef22-43c2-11e8-9621-83988df10f0c.jpg)
I want the scrren is second , no stats bar , no keyback button

**Comment by 544531645** [comment]
but sometime have ,somtime no have ... like point clound sometime is white

**Comment by 544531645** [comment]
1. if I delete Drawable drawable = getActivity().getResources().getDrawable(R.drawable.point_colud);
Bitmap bitmap = drawableToBitmap(drawable);
Texture chineseCharTexture = new Texture(bitmap, Texture.Format.RGBA8, false, true);
Material material1 = new Material();
material1.setDiffuseTexture(chineseCharTexture);
Surface surface = new Surface(0.25f, 0.25f);
surface.setMaterials(Arrays.asList(material1));
these code ,
use defalut point ,  it not slow

**Comment by VikAdvani** [comment]
Hi @544531645, thanks for reporting this issues. I've adding them to our backlog and will update this github issue when they are fixed.

**Comment by dthian** [comment]
This has been fixed in the latest release of ViroCore 1.7.2. Please checkout the update instructions here: https://virocore.viromedia.com/docs/releases
